aptos-move/framework/aptos-framework/sources/account.spec.move

```diff
@@ -1,6 +1,146 @@
+    }
+
+    spec get_authentication_key {
+        aborts_if !exists<Account>(addr);
+        ensures result == global<Account>(addr).authentication_key;
+    }
+
+    spec rotate_authentication_key_internal {
+        let addr = signer::address_of(account);
+        let account_resource = global<Account>(addr);
+        aborts_if !exists<Account>(addr);
+        aborts_if vector::length(new_auth_key) != 32;
+    }
+
+    spec assert_valid_signature_and_get_auth_key {
+        pragma aborts_if_is_partial;
+        aborts_if scheme != ED25519_SCHEME && scheme != MULTI_ED25519_SCHEME;
+    }
+
+    spec rotate_authentication_key {
+        pragma aborts_if_is_partial;
+        let addr = signer::address_of(account);
+        let account_resource = global<Account>(addr);
+        aborts_if !exists<Account>(addr);
+        aborts_if from_scheme != ED25519_SCHEME && from_scheme != MULTI_ED25519_SCHEME;
+    }
+
+    spec offer_signer_capability {
+        pragma aborts_if_is_partial;
+        let source_address = signer::address_of(account);
+        aborts_if !exists<Account>(recipient_address);
+        aborts_if account_scheme != ED25519_SCHEME && account_scheme != MULTI_ED25519_SCHEME;
+    }
+
+    spec revoke_signer_capability {
+        pragma aborts_if_is_partial;
+        aborts_if !exists<Account>(to_be_revoked_address);
+        ensures exists<Account>(to_be_revoked_address);
+    }
+
+    spec create_authorized_signer {
+        pragma aborts_if_is_partial;
+        let account_resource = borrow_global<Account>(offerer_address);
+        aborts_if !exists<Account>(offerer_address);
+        ensures exists<Account>(offerer_address);
+        ensures signer::address_of(result) == offerer_address;
+    }
+
+    spec create_resource_address {
+        pragma verify = false;
+    }
+
+    spec create_resource_account {
+        pragma verify = false;
+    }
+
+    spec create_framework_reserved_account {
+        pragma aborts_if_is_partial;
+        aborts_if addr != @0x1 &&
+                addr != @0x2 &&
+                addr != @0x3 &&
+                addr != @0x4 &&
+                addr != @0x5 &&
+                addr != @0x6 &&
+                addr != @0x7 &&
+                addr != @0x8 &&
+                addr != @0x9 &&
+                addr != @0xa;
+        ensures result_2 == SignerCapability { account: addr };
+    }
+
+    spec create_guid {
+        let addr = signer::address_of(account_signer);
+        let account = borrow_global_mut<Account>(addr);
+        aborts_if !exists<Account>(addr);
+        aborts_if account.guid_creation_num + 1 > MAX_U64;
+    }
+
+    spec new_event_handle {
+        let addr = signer::address_of(account);
+        let account = borrow_global_mut<Account>(addr);
+        aborts_if !exists<Account>(addr);
+        aborts_if account.guid_creation_num + 1 > MAX_U64;
+    }
+
+    spec register_coin {
+        pragma aborts_if_is_partial;
```

Comment:
I am not sure whether it is good idea, but if you add "aborts_if [abstract] false;" to the spec of type_of, aborts_if_is_partial is not needed here.

Reply:
done